dspace-xmlui/src/main/webapp/themes/Mirage/lib/xsl/aspect/sciencedirect/elsevier-embed.xsl

```diff
@@ -0,0 +1,62 @@
+<!--
+
+    The contents of this file are subject to the license and copyright
+    detailed in the LICENSE and NOTICE files at the root of the source
+    tree and available online at
+
+    http://www.dspace.org/license/
+
+-->
+<xsl:stylesheet
+        xmlns:i18n="http://apache.org/cocoon/i18n/2.1"
+        xmlns:dri="http://di.tamu.edu/DRI/1.0/"
+        xmlns:mets="http://www.loc.gov/METS/"
+        xmlns:dim="http://www.dspace.org/xmlns/dspace/dim"
+        xmlns:xlink="http://www.w3.org/TR/xlink/"
+        xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0"
+        xmlns="http://www.w3.org/1999/xhtml"
+        xmlns:xalan="http://xml.apache.org/xalan"
+        xmlns:encoder="xalan://java.net.URLEncoder"
+        xmlns:util="org.dspace.app.xmlui.utils.XSLUtils"
+        xmlns:jstring="java.lang.String"
+        xmlns:rights="http://cosimo.stanford.edu/sdr/metsrights/"
+        xmlns:confman="org.dspace.core.ConfigurationManager"
+        exclude-result-prefixes="xalan encoder i18n dri mets dim xlink xsl util jstring rights confman">
+
+    <xsl:variable name="embedWidth" select="confman:getProperty('elsevier-sciencedirect', 'embed.display.width')"/>
+    <xsl:variable name="embedHeight" select="confman:getProperty('elsevier-sciencedirect', 'embed.display.height')"/>
+
+    <xsl:template match="dri:div[@n='ElsevierEmbed']">
+        <iframe frameborder="0" scrolling="no" marginheight="0" marginwidth="0">
+            <xsl:attribute name="width">
+                <xsl:choose>
+                    <xsl:when test="string-length($embedWidth)=0">
+                        <xsl:text>700</xsl:text>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <xsl:value-of select="$embedWidth"/>
+                    </xsl:otherwise>
+                </xsl:choose>
+            </xsl:attribute>
+            <xsl:attribute name="height">
+                <xsl:choose>
+                    <xsl:when test="string-length($embedHeight)=0">
+                        <xsl:text>500</xsl:text>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <xsl:value-of select="$embedHeight"/>
+                    </xsl:otherwise>
+                </xsl:choose>
+            </xsl:attribute>
+            <xsl:attribute name="src">
+                <xsl:text>//api.elsevier.com/content/article/pii/</xsl:text>
```

Comment:
Hi Lieven,

This seems like good advice to me, do you agree?

Best regards,

Letitia Mukherjee
Market Development Manager Sharing Platforms
Mob: +31 6 13 98 98 71

From: Tim Donohue [mailto:notifications@github.com]
Sent: 26 January 2016 15:58
To: DSpace/DSpace
Cc: Mukherjee, Letitia (ELS-AMS)
Subject: Re: [DSpace] DS-2877 Import of ScienceDirect metadata including embargo and linking to or embedding of the final version (#1162)


In dspace-xmlui/src/main/webapp/themes/Mirage/lib/xsl/aspect/sciencedirect/elsevier-embed.xsl<https://github.com/DSpace/DSpace/pull/1162#discussion_r50844573>:

> +                        <xsl:value-of select="$embedWidth"/>

> +                    </xsl:otherwise>

> +                </xsl:choose>

> +            </xsl:attribute>

> +            <xsl:attribute name="height">

> +                <xsl:choose>

> +                    <xsl:when test="string-length($embedHeight)=0">

> +                        <xsl:text>500</xsl:text>

> +                    </xsl:when>

> +                    <xsl:otherwise>

> +                        <xsl:value-of select="$embedHeight"/>

> +                    </xsl:otherwise>

> +                </xsl:choose>

> +            </xsl:attribute>

> +            <xsl:attribute name="src">

> +                <xsl:text>//api.elsevier.com/content/article/pii/</xsl:text>

This URL path should be pulled in via configuration instead of hardcoded (similar to how the 'api.key' is pulled in via configuration below). That way if it changes, we can tweak its configuration easily.

—
Reply to this email directly or view it on GitHub<https://github.com/DSpace/DSpace/pull/1162/files#r50844573>.

________________________________

Elsevier B.V. Registered Office: Radarweg 29, 1043 NX Amsterdam, The Netherlands, Registration No. 33156677, Registered in The Netherlands.